weather.py

```diff
@@ -1,104 +1,426 @@
-        weather_data: A list of numbers.
-    Returns:
-        The minium value and it's position in the list.
-    """
-    pass
-
-
-def find_max(weather_data):
-    """Calculates the maximum value in a list of numbers.
-
-    Args:
-        weather_data: A list of numbers.
-    Returns:
-        The maximum value and it's position in the list.
-    """
-    pass
-
-
-def generate_summary(weather_data):
-    """Outputs a summary for the given weather data.
-
-    Args:
-        weather_data: A list of lists, where each sublist represents a day of weather data.
-    Returns:
-        A string containing the summary information.
-    """
-    pass
-
-
-def generate_daily_summary(weather_data):
-    """Outputs a daily summary for the given weather data.
-
-    Args:
-        weather_data: A list of lists, where each sublist represents a day of weather data.
-    Returns:
-        A string containing the summary information.
-    """
-    pass
+import csv
+from datetime import datetime
+
+DEGREE_SYBMOL = u"\N{DEGREE SIGN}C"
+
+def format_temperature(temp):
+    """Takes a temperature and returns it in string format with the degrees
+        and celcius symbols.
+
+    Args:
+        temp: A string representing a temperature.
+    Returns:
+        A string contain the temperature and "degrees celcius."
+    """
+    return f"{temp}{DEGREE_SYBMOL}"
+
+'''
+Test examples below for format_temperature:
+    print(format_temperature(1))
+    print(format_temperature(10))
+'''
+
+
+## CHALLENGE 1
+def convert_date(iso_string):
+    """Converts and ISO formatted date into a human readable format.
+
+    Args:
+        iso_string: An ISO date string..
+    Returns:
+        A date formatted like: Weekday Date Month Year e.g. Tuesday 06 July 2021
+    """
+
+    date = datetime(int(iso_string[0:4]),int(iso_string[5:7]),int(iso_string[8:10]))
+    return date.strftime(f"%A %d %B %Y")
+
+#print(convert_date("2021-07-02T07:00:00+08:00"))
+
+''' Test cases to put into 'convert_date' to see if the function works 
+    1. 2021-07-02T07:00:00+08:00
+    2. 2010-01-27T07:00:00+08:00
+    3. 2030-12-25T07:00:00+08:00
+'''
+
+''' 
+Annie's working out paper below:
+    > iso_string = datetime(2001,1,1)
+    > return iso_string.strftime(f"%A %d %B %Y")
+
+#curious about: input = (input(write your year-month-day)) > if year-month-day is 2001-12-29 > the date value below will read the year-month-date and this will be the input into the function. Code for this below:
+    > iso_string = input("write your date in year/month/day eg. 2001/12/29 ")
+    > print(convert_date(iso_string))
+# '''
+
+
+### CHALLENGE 2
+def convert_f_to_c(temp_in_farenheit):
+    """Converts an temperature from farenheit to celcius.
+
+    Args:
+        temp_in_farenheit: float representing a temperature.
+    Returns:
+        A float representing a temperature in degrees celcius, rounded to 1dp.
+    """
+    x = (float(temp_in_farenheit) - 32) * 5/9
+    x = round(x,1)
+    return x

```

Comment:
This works, but it's generally good practice to use an informative variable name to replace `x` in these 2 lines (e.g. `temp_in_c`, `rounded_temp_in_c`). The use of `x` here is an example of 'Magic numbers', which can make code unreadable if there are multiple unnamed constants throughout.

https://en.m.wikipedia.org/wiki/Magic_number_(programming)#:~:text=The%20term%20magic%20number%20or,1%20manuals%20of%20the%201960s